aten/src/ATen/Dispatch.h

```diff
@@ -341,6 +341,27 @@ inline void deprecated_AT_DISPATCH_ALL_TYPES_AND_HALF_AND_COMPLEX() {}
           at::kQUInt8, at::quint8, at::kByte, uint8_t, __VA_ARGS__)         \
       AT_QINT_PRIVATE_CASE_TYPE(                                            \
           at::kQInt32, at::qint32, at::kInt, int, __VA_ARGS__)              \
+      AT_QINT_PRIVATE_CASE_TYPE(                                            \
+          at::kQUInt4x2, at::quint8, at::kByte, uint8_t, __VA_ARGS__)       \
+      default:                                                              \
+        AT_ERROR(#NAME, " not implemented for '", toString(TYPE), "'");     \
+    }                                                                       \
+  }()
+
+#define AT_DISPATCH_QINT_TYPES_SUB_BYTE(TYPE, NAME, ...)                    \
+  [&] {                                                                     \
+    const auto& the_type = TYPE;                                            \
+    /* don't use TYPE again in case it is an expensive or side-effect op */ \
+    at::ScalarType _st = ::detail::scalar_type(the_type);                   \
+    switch (_st) {                                                          \
+      AT_QINT_PRIVATE_CASE_TYPE(                                            \
+          at::kQInt8, at::qint8, at::kChar, int8_t, __VA_ARGS__)            \
+      AT_QINT_PRIVATE_CASE_TYPE(                                            \
+          at::kQUInt8, at::quint8, at::kByte, uint8_t, __VA_ARGS__)         \
+      AT_QINT_PRIVATE_CASE_TYPE(                                            \
```

Comment:
For sub_byte dispatch, wondering why kQInt32 is being considered?

Reply:
I need to clean this up, this is just a prototype. But this will also have bit rate as an extra field in addition to underlying type storage.